anthos-bm-edge-deployment/.gitignore

```diff
@@ -0,0 +1,99 @@
+target/
+pkg/
+tmp/
+
+.gitcookies
+
+.screenrc
+
+# Newrelic agent log
+newrelic_agent.log
+
+# test
+
+# Folder to hold build-artifacts needed for Docker install (maybe all)
+# keep the .gitkeep file, but ignore everything else
+build-artifacts/*
+!build-artifacts/README.md
+!build-artifacts/ssh-config
+
+# local Inventory files are not stored
+inventory.yml
+inventory.yaml
+# GCP inventory files generated
+gcp.yaml
+gcp.yml
+
+inventory/*.yaml
+
+# never check in any public keys
+*.pub
+
+# Downloaded git repo
+serving/
+# Downloaded model
+models/**/
+
+# Debugging & Testing Ansible
+build.yml
+build.yaml
+roles/test-*/
+
+# Molecule
+.cache/
+
+# Generated configuration file
+k8s/overlays/**/configuration/*-hydrated.conf
+k8s/overlays/**/configuration/secrets/*.json
+
+### VS Code ###
+.vscode/
+
+### IntelliJ ###
+.idea/
```

Comment:
Nitpick: 
This might be out-of-scope for this PR.
But we should pull all broadly applicable `.gitignore` entries (such as `.idea/`) to the root `/.gitignore`.
There is already some duplication (e.g., `.DS_Store` and `.vscode` exist in both).